src/armor_layers.cpp

```diff
@@ -1031,6 +1034,36 @@ void outfit::sort_armor( Character &guy )
                     tmp_worn[leftListIndex]->unset_flag( json_flag_HIDDEN );
                 }
             }
+        } else if( action == "CHANGE_SPRITE" && leftListIndex < leftListSize ) {
+            uilist menu;
+            menu.title = _( "Change sprite" );
+            menu.addentry( 0, true, MENU_AUTOASSIGN, _( "Select sprite from items" ) );
+            menu.addentry( 1, true, MENU_AUTOASSIGN, _( "Restore default sprite" ) );
+            menu.addentry( 2, true, MENU_AUTOASSIGN, _( "Cancel" ) );
+            menu.query();
+            if( menu.ret == 0 ) {
+                item_location loc;
+                avatar *you = player_character.as_avatar();
+                auto armor_filter = [&]( const item & i ) {
+                    return i.is_armor();
+                };
+                if( you != nullptr ) {
+                    loc = game_menus::inv::titled_filter_menu( armor_filter,
+                            *you,
+                            _( "Select appearance of this armor:" ),
+                            -1,
+                            _( "You have nothing to wear." ) );
+                }
+                if( loc && loc.get_item() ) {
+                    const item *i = loc.get_item();
+                    const std::string variant = i->has_itype_variant() ? i->itype_variant().id : "";
+                    tmp_worn[leftListIndex]->set_var( "sprite_override", i->typeId().str() );
+                    tmp_worn[leftListIndex]->set_var( "sprite_override_variant", variant );
+                }
+            } else if( menu.ret == 1 ) {
+                tmp_worn[leftListIndex]->erase_var( "sprite_override" );
+                tmp_worn[leftListIndex]->erase_var( "sprite_override_variant" );
+            }
```

Comment:
Might be just a drop in the ocean, but it'd be great if you would extract that in its own method, `outfit::sort_armor` is already far too big as is.

Reply:
Should I make new function like outfit::change_sprite, or make new interface outside sorting armor menu?

Reply:
It doesn't have to be a member of the outfit class, a static method in the same file is enough. Unless you have some plans to make this functionality available from other places, too.